bin/reth/src/test_eth_chain/runner.rs

```diff
@@ -1,13 +1,161 @@
+                ForkSpec::Constantinople |
+                ForkSpec::MergeEOF |
+                ForkSpec::MergeMeterInitCode
+        ) {
+            continue
+        }
+
+        if matches!(suite.pre, State(RootOrState::Root(_))) {}
+
+        let pre_state = match suite.pre.0 {
+            RootOrState::State(state) => state,
+            RootOrState::Root(_) => {
+                info!("Skipping test {name}...");
+                continue
+            }
+        };
+
+        debug!("Executing test: {name} for spec: {:?}", suite.network);
+
+        let spec_upgrades: SpecUpgrades = suite.network.into();
+        // if paris aka merge is not activated we dont have block rewards;
+        let has_block_reward = spec_upgrades.paris != 0;
+
+        // Create db and acquire transaction
+        let db = create_test_rw_db::<WriteMap>();
+        let tx = db.tx_mut()?;
+
+        // insert genesis
+        let header: SealedHeader = suite.genesis_block_header.into();
+        let genesis_block = BlockLocked { header, body: vec![], ommers: vec![] };
+        reth_provider::insert_canonical_block(&tx, &genesis_block, has_block_reward)?;
+
+        suite.blocks.iter().try_for_each(|block| -> eyre::Result<()> {
+            let decoded = BlockLocked::decode(&mut block.rlp.as_ref())?;
+            reth_provider::insert_canonical_block(&tx, &decoded, has_block_reward)?;
+            Ok(())
+        })?;
+
+        pre_state.into_iter().try_for_each(|(address, account)| -> eyre::Result<()> {
+            let has_code = !account.code.is_empty();
+            let code_hash = if has_code { Some(keccak256(&account.code)) } else { None };
+            tx.put::<tables::PlainAccountState>(
+                address,
+                RethAccount {
+                    balance: account.balance.0,
+                    nonce: account.nonce.0.as_u64(),
+                    bytecode_hash: code_hash,
+                },
+            )?;
+            if let Some(code_hash) = code_hash {
+                tx.put::<tables::Bytecodes>(code_hash, account.code.to_vec())?;
+            }
+            account.storage.iter().try_for_each(|(k, v)| {
+                tx.put::<tables::PlainStorageState>(
+                    address,
+                    StorageEntry { key: H256::from_uint(&k.0), value: v.0 },
+                )
+            })?;
+
+            Ok(())
+        })?;
+
+        // Commit the pre suite state
+        tx.commit()?;
+
+        // Initialize the execution stage
+        // Hardcode the chain_id to Ethereums 1.
+        let mut stage =
+            ExecutionStage::new(reth_executor::Config { chain_id: 1.into(), spec_upgrades });
+
+        // Call execution stage
+        let input = ExecInput::default(); // TODO:
+        stage.execute(&mut StageDB::new(db.as_ref())?, input).await?;
 
-    for suit in suits.0 {
-        println!("TODO:{:?}", suit.0);
+        // Validate post state
+        // TODO:
```

Comment:
What is the expected post-state check to add here?

Reply:
Use `suit.post_state` and compare it with database.